launcher/container_runner.go

```diff
@@ -532,25 +540,53 @@ func initImage(ctx context.Context, cdClient *containerd.Client, launchSpec spec
 	return image, nil
 }
 
-func getImageLabels(ctx context.Context, image containerd.Image) (map[string]string, error) {
-	// TODO(jiankun): Switch to containerd's WithImageConfigLabels()
+// openPorts writes firewall rules to accept all traffic into that port and protocol using iptables.
+func openPorts(ports map[string]struct{}) error {
+	for k := range ports {
+		portAndProtocol := strings.Split(k, "/")
+		if len(portAndProtocol) != 2 {
+			return fmt.Errorf("failed to parse port and protocol, expect [port]/[protocol] like 80/tcp, got %s", portAndProtocol)
+		}
+
+		port := portAndProtocol[0]
+		_, err := strconv.ParseUint(port, 10, 16)
+		if err != nil {
+			return fmt.Errorf("failed to parse the port as an integer: %w", err)
+		}
+
+		protocol := portAndProtocol[1]
+		if protocol != "tcp" && protocol != "udp" {
+			return fmt.Errorf("unrecognized protocol, expect tcp or udp, got %s", protocol)
+		}
+
+		cmd := exec.Command("iptables", "-A", "INPUT", "-p", protocol, "--dport", port, "-j", "ACCEPT")
```

Comment:
While this can be straightforwardly parsed from reading the manpage, it's a good to write a quick explanation on why we pick the options specifically and some assumption.